### PR TITLE
Add globbing pattern instructions to quick start guide

### PR DESCRIPTION
**Pull request by apoorvedave1**
Add instructions on how to set and use globbing patterns in index creation and maintenance.

### What is the context for this pull request?

 - **Tracking Issue**: #283
 - **Parent Issue**: #226, #269

### What changes were proposed in this pull request?
Documentation update: Add instructions on using globbing pattern

### CODE REVIEW

File: docs/_docs/02-ug-configuration.md [hunk 1]
```diff
@@ -19,3 +19,4 @@ classes: wide
...
 | spark.hyperspace.index.hybridscan.delete.enabled   | false                                                                                    | (temporary) Enable Hybrid Scan for deleted files. Lineage is required to handle deleted files in Hybrid Scan.  | 0.3.0         |
 | spark.hyperspace.index.hybridscan.delete.maxNumDeleted   | 30                                                                                    | Threshold of the maximum number of deleted files for Hybrid Scan. | 0.3.0         |
+| spark.hyperspace.source.globbingPattern   | N/A                                                                                    | Dataframe option (not a spark config) to allow indexes on globbing patterns. E.g. `spark.read.option("spark.hyperspace.source.globbingPattern", "/temp/*/*").parquet("/temp/*/*")` | 0.4.0         |
```
Comment (imback82) [inline docs/_docs/02-ug-configuration.md:22]: You mean `DataFrameReader` option?

**Review by imback82 — COMMENTED** [review]
LGTM (one comment), thanks @apoorvedave1!

File: docs/_docs/02-ug-configuration.md (hunk 1 above)
Comment (apoorvedave1) [inline docs/_docs/02-ug-configuration.md:22]:
```suggestion
| spark.hyperspace.source.globbingPattern   | N/A                                                                                    | DataFrameReader option (not a spark config) to allow indexes on globbing patterns. E.g. `spark.read.option("spark.hyperspace.source.globbingPattern", "/temp/*/*").parquet("/temp/*/*")` | 0.4.0         |
```

Comment (apoorvedave1) [reply docs/_docs/02-ug-configuration.md]: thanks @imback82 , done

**Review by imback82 — APPROVED** [review]

**Pull request merged**